src/java/org/jivesoftware/openfire/plugin/util/cluster/ClusterPacketRouter.java

```diff
@@ -40,9 +41,11 @@
     public boolean routePacket(byte[] nodeID, JID receipient, Packet packet) {
         // Send the packet to the specified node and let the remote node deliver the packet to the recipient
         try {
+            if (!ClusterManager.isClusterMember(nodeID))
+                return false;
             CacheFactory.doClusterTask(new RemotePacketExecution(receipient, packet), nodeID);
             return true;
-        } catch (IllegalStateException  e) {
+        } catch (IllegalArgumentException | IllegalStateException  e) {
             logger.warn("Error while routing packet to remote node: " + e);
```

Comment:
And I know you didn't change this, but reviewing your changes has highlighted that this should be
```java
 logger.warn("Error while routing packet to remote node: ", e);
```
so the stack trace isn't lost.

Reply:
done